pennylane/transforms/decompose.py

```diff
@@ -98,12 +151,31 @@ def __init__(self, gate_set=None, max_expansion=None):
                 gate_set = set([gate_set])
 
             if isinstance(gate_set, Iterable):
-                gate_types = tuple(gate for gate in gate_set if isinstance(gate, type))
-                gate_names = set(gate for gate in gate_set if isinstance(gate, str))
-                self.gate_set = lambda op: (op.name in gate_names) or isinstance(op, gate_types)
+                target_gate_types = tuple(gate for gate in gate_set if isinstance(gate, type))
+                target_gate_names = set(gate for gate in gate_set if isinstance(gate, str))
+                self.gate_set = lambda op: (op.name in target_gate_names) or isinstance(
+                    op, target_gate_types
+                )
+
+                if qml.decomposition.enabled_graph():
+
+                    if target_gate_types:
+                        raise ValueError(
+                            f"The graph-based decomposition doesn't support Operator types: {target_gate_types}."
+                        )
```

Comment:
Why raise an error here? Can't you collect all the gate names from `self.gate_set`?